src/main/java/net/minecraftforge/event/terraingen/DecorateBiomeEvent.java

```diff
@@ -96,27 +98,94 @@ public Post(World world, Random rand, BlockPos pos)
 
     /**
      * This event is fired when a chunk is decorated with a biome feature.
-     *
+     * The amount denotes how many instances of the feature will be generated.
+     * <p>
      * You can set the result to DENY to prevent the default biome decoration.
      */
     @HasResult
     public static class Decorate extends DecorateBiomeEvent
     {
+        private final EventType type;
+        @Nullable
+        private final Integer totalAmount;
+        @Nullable
+        private Integer modifiedAmount;
+
```

Comment:
So the only real thing you care about in this PR is being able to say 'I want you to spawn 10 trees instead of 5!'
Thats fine and danda, if thats the case then don't do it with this event.
The hack around null Integers is dumb.
Just make a sub event for 'countable' {there has to be a better name} types. Ones that are fired in loops and fire them. 
I don't like the concept of polling the numerical value from an event but I can KINDA see your point. 
But it doesn't solve or even address the issue of conflicting handlers A wants 10, B wants 2, C wants to cancel. Who wins?
A wants 10, B wants 2, who wins?